core/src/test/java/com/linecorp/armeria/common/util/CompositeExceptionTest.java

```diff
@@ -0,0 +1,257 @@
+        final IllegalStateException ex1 = new IllegalStateException();
+        final IllegalArgumentException ex2 = new IllegalArgumentException();
+        final String className = getClass().getSimpleName();
+        final String methodName = "verboseExceptionDisabledTestIfStackTraceLengthLessThan20";
+        final String fileName = "CompositeExceptionTest.java";
+        final StackTraceElement[] customStackTrace = {
+                new StackTraceElement(className, methodName, fileName, 1),
+                new StackTraceElement(className, methodName, fileName, 2),
+                new StackTraceElement(className, methodName, fileName, 3),
+        };
+        ex1.setStackTrace(customStackTrace);
+        ex2.setStackTrace(customStackTrace);
+        final CompositeException compositeException =
+                new CompositeException(ImmutableList.of(ex1, ex2), Sampler.never());
+        final int paddingCount = Arrays.stream(compositeException.getCause().getMessage().split(separator))
+              .filter(line -> !line.contains("Multiple exceptions") && !line.contains("|-- "))
+              .mapToInt(line -> {
+                  int padding = 0;
+                  final int paddingLength = PADDING.length();
+                  while (line.startsWith(PADDING)) {
+                      line = line.substring(paddingLength);
+                      padding += paddingLength;
+                  }
+                  return padding;
+              })
+              .sum();
+
+        // padding * stack trace length * exception length
+        final int expectPaddingCount = 4 * 3 * 2;
+        assertThat(paddingCount).isEqualTo(expectPaddingCount);
+    }
+
+    @Test
+    public void compositeExceptionPaddingTest() {
+        final IllegalStateException ex1 = new IllegalStateException();
+        final IllegalArgumentException ex2 = new IllegalArgumentException();
+        final AnticipatedException ex3 = new AnticipatedException(ex2);
+        final String className = getClass().getSimpleName();
+        final String methodName = "verboseExceptionDisabledTestIfStackTraceLengthLessThan20";
+        final String fileName = "CompositeExceptionTest.java";
+        final StackTraceElement[] customStackTrace = {
+                new StackTraceElement(className, methodName, fileName, 1),
+                new StackTraceElement(className, methodName, fileName, 2),
+                new StackTraceElement(className, methodName, fileName, 3),
+        };
+        ex1.setStackTrace(customStackTrace);
+        ex2.setStackTrace(customStackTrace);
+        ex3.setStackTrace(customStackTrace);
+        final CompositeException compositeException =
+                new CompositeException(ImmutableList.of(ex1, ex3), Sampler.never());
+        final int paddingCount = Arrays.stream(compositeException.getCause().getMessage().split(separator))
+             .filter(line -> !line.contains("Multiple exceptions") && !line.contains("|-- "))
```

Comment:
The result of the padding is now:
```
|-- com.linecorp.armeria.internal.testing.AnticipatedException: java.lang.IllegalArgumentException
    at CompositeExceptionTest.verboseExceptionDisabledTestIfStackTraceLengthLessThan20(CompositeExceptionTest.java:1)
    ...
  |-- java.lang.IllegalArgumentException: null
```
Shoudn't it be this?
```
|-- com.linecorp.armeria.internal.testing.AnticipatedException: java.lang.IllegalArgumentException
    at CompositeExceptionTest.verboseExceptionDisabledTestIfStackTraceLengthLessThan20(CompositeExceptionTest.java:1)
    ...
    |-- java.lang.IllegalArgumentException: null
        at ...
```

Reply:
yes!
```java
if (!seenCauses.containsKey(inner)) {
    seenCauses.put(inner, true);

    inner = inner.getCause();
    depth += 2;
} else {
```
If so, it should be like this.
that's correct?

Reply:
<img width="1192" alt="스크린샷 2022-08-01 오후 1 19 30" src="https://user-images.githubusercontent.com/23313008/182071474-25e20708-b631-4afe-a42d-0524feab9a38.png">

Reply:
But wouldn't that change the variable(variable name?) significantly from its original intention?

Reply:
Oops, sorry for making you get confused. 😓 
I have checked the current implementation and it just adds two spaces as the depth is increased and I think it's fine as is. 😄